pkg/user/signer_test.go

```diff
@@ -114,6 +114,18 @@ func (s *SignerTestSuite) TestConfirmTx() {
 	})
 }
 
+func (s *SignerTestSuite) TestGasEstimation() {
+	msg := bank.NewMsgSend(s.signer.Address(), testnode.RandomAddress().(sdk.AccAddress), sdk.NewCoins(sdk.NewInt64Coin(app.BondDenom, 10)))
+	seq := s.signer.Sequence()
+	txBytes, err := s.signer.CreateTx([]sdk.Msg{msg})
+	require.NoError(s.T(), err)
+	// reset the sequence
+	s.signer.ForceSetSequence(seq)
```

Comment:
[question] why does the sequence need to be reset? Based on the `EstimateGas` function body, it seems like it should be possible to estimate the gas for a tx regardless of if the sequence is 0 or not.

```suggestion
```

[tangent] if it is important to test that estimate gas works when sequence == 0 and sequence != 0, perhaps two tests cases would make sense here.

Reply:
Hmm maybe I could rewrite the `EstimateGas` function signature so instead of txBytes (which need to be generated using `CreateTx` and which automatically increment the sequence) to only take an array of `sdk.Msgs` and option types and create an unsigned transaction that way. What do you think?

Reply:
> why does the sequence need to be reset?

Because `CreateTx` automatically increments the sequence

Reply:
Hmm that refactor idea seems okay to me but I don't follow why the refactor is needed. If I remove the line to `ForceSetSequence(seq)`, the test still passes.

Reply:
> If I remove the line to ForceSetSequence(seq), the test still passes.

Really? It shouldn't. You should get a sequence mismatch for the following test in the suite

Reply:
oh I only ran that individual test not the remainder of the test suite. It seems like a code smell if other tests break based on this tests side-effects. Definitely not blocking on it but I'm open to your refactor idea if that means tests won't impact each other in this suite.

Reply:
The reason for refactoring is more a matter of better dev UX. it's a bit of a foot gun if creating the tx you need to estimate the gas increments the sequence number